payments/layer-2-payments.md

```diff
@@ -0,0 +1,62 @@
+
+zkSync supports close-to-immediate transactions involving both ETH itself and potentially any ERC-20-compatible token. Additionally, the transactions fees can be paid in the token that's transferred, eliminating the need for the sender to be in posession of ETH.
+
+Thanks to the underlying technology, namely zkRollup, the validators in zkSync are able to process thousands of transactions in a single block. The hash of such a block and a cryptographic proof that the block is a result of an application of a series of correct state transitions is published to Ethereum blockchain. Alongside it, the delta change resultant from execution of all the included transactions is also published to the blockchain and both the delta and the proof are validated by a smart contract.
+
+The net effect is that, with significantly smaller costs and much higher transaction throughput, zkSync remains almost as secure as the Ethereum mainnet itself.
+
+### Polygon
+
+Formely dubbed "Matic", Polygon is becoming a de-facto standard for a lot of decentralized apps in the Ethereum ecosystem. Technically, it could be argued that it's not really a "Layer 2" solution as it doesn't keep in sync with the Ethereum but rather maintains its own side chain, connected to the Ethereum main chain with a set of inter-chain bridges.
+
+What makes Polygon attractive is that its chain is fully compatible with the Ethereum, making all the smart contracts, wallets and other applications currently working with Ethereum immediately usable on Polygon. One noticeable difference is that in place of the ETH token, the Polygon chain uses its own MATIC token to pay the transaction fees.
+
+The consensus on Polygon's mainnet chain is maintained by a carefully selected and somewhat decentralized set of validator nodes who use a proof-of-stake mechanism, whereby the validators stake their MATIC tokens. Because of this set-up and because no active synchronization with the Ethereum is attempted, the costs of transactions are several orders of magnitude lower than on Ethereum.
+
+A user wishing to enter and exit Polygon - that is transfer funds between Polygon and the Ethereum mainnet - needs to use a bridge - a special kind of contract that locks user's funds in one chains and issues them a respective amount in the other chain. Because those bridges need to work on the Ethereum, they're pretty costly to interact with. The good news is that a user having funds on Polygon may not have an immediate need to exit from Polygon to mainnet as there are a lot of centralized and decentralized exchanges and other kinds of DeFi solutions already operating directly _on_ the Polygon chain.
```

Comment:
```suggestion
A user wishing to enter and exit Polygon - that is transfer funds between Polygon and the Ethereum mainnet - needs to use a bridge - a special kind of contract that locks user's funds in one chain and issues them a respective amount in the other chain. Because those bridges need to work on the Ethereum, they're pretty costly to interact with. The good news is that a user having funds on Polygon may not have an immediate need to exit from Polygon to mainnet as there are a lot of centralized and decentralized exchanges and other kinds of DeFi solutions already operating directly _on_ the Polygon chain.
```

Reply:
ah, I already noticed the mistake and corrected it